README.md

```diff
@@ -43,13 +43,34 @@ npm test
 
 ## Building from source
 
-Compiled from https://github.com/facebook/zstd/tree/dev/contrib/single_file_libs, with the
+Compiled from https://github.com/facebook/zstd/tree/dev/build/single_file_libs, with the
 following steps:
 
 ```shell
 ./combine.sh -r ../../lib -o zstddeclib.c zstddeclib-in.c
-emcc zstddeclib.c -Oz -s EXPORTED_FUNCTIONS="['_ZSTD_decompress', '_ZSTD_findDecompressedSize', '_ZSTD_isError', '_malloc', '_free']" -s ALLOW_MEMORY_GROWTH=1 -s MALLOC=emmalloc -o zstddec.wasm
-base64 zstddec.wasm > zstddec.txt
+```
+
+Add the following to the end of *zstddeclib.c* to [shim out unused wasi
+imports](https://github.com/emscripten-core/emscripten/issues/17331),
+
+```c
+int __wasi_fd_write(int, int, int, int) {
+  __builtin_trap();
+}
+int __wasi_fd_seek(int, int64_t, int, int) {
+  __builtin_trap();
+}
+int __wasi_fd_close(int) {
+  __builtin_trap();
+}
+_Noreturn void  __wasi_proc_exit(int) {
+  __builtin_trap();
+}
+```
+
+```shell
+emcc zstddeclib.c -Oz -s EXPORTED_FUNCTIONS="['_ZSTD_decompress', '_ZSTD_findDecompressedSize', '_ZSTD_isError', '_malloc', '_free']"  -Wl,--no-entry -s ALLOW_MEMORY_GROWTH=1 -s MALLOC=emmalloc -o zstddec.wasm
```

Comment:
Just for my own learning, what is `-Wl,--no-entry`? I couldn't find this in the `emcc` help output.

Reply:
It is a standard linking flag for not creating the entry function when the program starts up (what gets called before `main`). This is usually `_start`, which is also [true for WebAssembly](https://lld.llvm.org/WebAssembly.html), but we do not need it because we are calling the other functions instead, `_ZSTD_decompress`, etc.